[Docs] Add `PlotOutputs` example to VizroAi docs (#599)
mckinsey/vizro: vizro-ai/docs/pages/user-guides/run-vizro-ai.md

            ```

2. When the `return_elements` argument of VizroAI's `plot` method is set to `True`, the method returns a `PlotOutputs` data class which contains all possible `VizroAI.plot()` outputs.
By setting `return_elements=True` you can access code or the figure object. Vizro-AI validates the code to ensure that it is executable and can be integrated.

    !!! example "Accessing outputs from PlotOutputs data class"

        === "app.py"
           ```py
           import vizro_ai
           from vizro_ai import VizroAI
           import vizro.plotly.express as px
           from dotenv import load_dotenv

           load_dotenv()

           df = px.data.gapminder()
           vizro_ai = VizroAI()

           plot_outputs = vizro_ai.plot(df, "describe life expectancy per continent over time", explain=True, return_elements=True)
           fig = plot_outputs.figure
           code_string = plot_outputs.code
           ```

> Alternatively, you can access code by Vizro-AI's `_get_chart_code` method which returns a string of Python code that manipulates the data and creates the visualization.

Reviewer: Add a note saying this is deprecated. please use .code instead. Even just remove it?

> Alternatively, you can access code by Vizro-AI's `_get_chart_code` method which returns a string of Python code that manipulates the data and creates the visualization.

Reviewer 2: ```suggestion
You can also access code with Vizro-AI's `_get_chart_code` method, which returns a string of Python code that manipulates the data and creates the visualization.
```

Reviewer 2: I would have expected Vale to flag the word "alternatively" to warn you it's more complex than using "also" but it obviously hasn't. There are a couple of other places I'd have thought it would warn you (the word encapsulate for example) but as it hasn't, i'll wait until this merges and spend some time in the next few days looking to see why it's not running as expected. Don't want to hold you up and this text is fine as is anyway, even if some usage would be considered more complex than typical.

!!! example "Application integration via chart code"

    === "app.py"
        ```py
        import vizro.plotly.express as px
        from vizro_ai import VizroAI

        vizro_ai = VizroAI()

        df = px.data.gapminder()
        code_string = vizro_ai._get_chart_code(df, "describe life expectancy per continent over time")
        ```
    === "code_string"

>         [![ResultCode]][ResultCode]

Reviewer 3: Should we not remove this?

To use the insights or code explanation, you can use `vizro_ai._run_plot_tasks(df, ..., explain=True)`, which returns a dictionary containing the code explanation and chart insights alongside the code.

### How to use `max_debug_retry` parameter in plot function
- Default Value: 3
- Type: `int`
- Brief: By default, the `max_debug_retry` is set to 3, the function will try to debug errors up to three times.
If the errors are not resolved after the maximum number of retries, the function will stop further debugging retries.
For example, if you would like adjust to 5 retries, you can set `max_debug_retry = 5` in the plot function:

```py
vizro_ai.plot(df = df, user_input = "your user input", max_debug_retry= 5)
```

### How to use `return_elements` parameter in plot function
- Default Value: False
- Type: `bool`
- Brief: By default, the `return_elements` is set to `False`, and the `VizroAI.plot()` returns a `plotly.graph_objects` object. If `return_elements` is set to `True` `VizroAI.plot()` returns the `PlotOutputs` data class.

> The `PlotOutputs` data class is designed to encapsulate all possible outputs generated by the `VizroAI.plot()` method. This class ensures that all outputs are organized and accessible.

Reviewer 3: ```suggestion
The `PlotOutputs` data class is designed to encapsulate all possible outputs generated by the `VizroAI.plot()` method.
```

Attributes of `PlotOutputs`:

- **`code`**: A string representing of the Python code that manipulates the data and creates the visualization.

> - **`figure`**: An instance of a Plotly `go.Figure` object, representing the visual plot generated by `VizroAI.plot()`.

Reviewer 3: Hm, technically it is a CapturedCallable.. Maybe we can say that and link to the Vizro docs, and explain that this is so it immediately runs in a dashboard. Otherwise it just behaves like a plotly `go.Figure`

- **`business_insights`**: A string containing high-level business insights derived from the plot. `business_insights` is only available if `explain=True`.
- **`code_explanation`**: A string offering a detailed explanation of the code used to produce the plot. `code_explanation` is only available if `explain=True`.

In the example below we aim to illustrate the use of `PlotOutputs` data class.

  ```py
  import vizro_ai
  from vizro_ai import VizroAI
  import vizro.plotly.express as px
  from dotenv import load_dotenv

  load_dotenv()

  df = px.data.gapminder()
  vizro_ai = VizroAI(model="gpt-4-0613")

  plot_outputs = vizro_ai.plot(df, explain=True,  return_elements=True)

  # accessing figure object from PlotOutputs
  plot_outputs.figure
  # accessing code string from PlotOutputs
  plot_outputs.code
  # accessing code explanation string from PlotOutputs
  plot_outputs.code_explanation
  # accessing business insights string from PlotOutputs
  plot_outputs.business_insights

>   ```

Reviewer 3: It hink we don't even need this no?
